filesystem/storage/DBFile.php

```diff
@@ -0,0 +1,457 @@
+<?php
+
+use SilverStripe\Filesystem\ImageManipulation;
+use SilverStripe\Filesystem\Storage\AssetContainer;
+use SilverStripe\Filesystem\Storage\AssetStore;
+
+// Un-comment once https://github.com/silverstripe/silverstripe-framework/pull/4551/ is merged
+// namespace SilverStripe\Filesystem\Storage;
+
+/**
+ * Represents a file reference stored in a database
+ *
+ * @property string $Hash SHA of the file
+ * @property string $Filename Name of the file, including directory
+ * @property string $Variant Variant of the file
+ *
+ * @package framework
+ * @subpackage model
+ */
+class DBFile extends CompositeDBField implements AssetContainer, ShortcodeHandler {
+
+	use ImageManipulation;
+
+	/**
+	 * List of allowed file categories.
+	 *
+	 * {@see File::$app_categories}
+	 *
+	 * @var array
+	 */
+	protected $allowedCategories = array();
+
+	/**
+	 * List of image mime types supported by the image manipulations API
+	 *
+	 * {@see File::app_categories} for matching extensions.
+	 *
+	 * @config
+	 * @var array
+	 */
+	private static $supported_images = array(
+		'image/jpeg',
+		'image/gif',
+		'image/png'
+	);
+
+	/**
+	 * Create a new image manipulation
+	 *
+	 * @param string $name
+	 * @param array|string $allowed List of allowed file categories (not extensions), as per File::$app_categories
+	 */
+	public function __construct($name = null, $allowed = array()) {
+		parent::__construct($name);
+		$this->setAllowedCategories($allowed);
+	}
+
+	/**
+	 * Determine if a valid non-empty image exists behind this asset, which is a format
+	 * compatible with image manipulations
+	 *
+	 * @return boolean
+	 */
+	public function getIsImage() {
```

Comment:
At the moment, the "can be displayed in an img tag" and "can be resized" are equivalent tests. We could make a new API for "get the tag of the arbitrary file", but we'd end up with a diverged purpose of the `Image` dataobject. As it stands, the expectations of it are pretty clear, and it would be awkward to muddy the definition. The `image` / `image/supported` is probably as awkward as I'd like the distinction to get, since even within non-supported images, not all are embeddable in HTML.